### PR TITLE
HPCC-27663 Mine checkbox on WU list page unclickable

### PR DESCRIPTION
**Pull request by jeclrsg**
Signed-off-by: Jeremy Clements <jeremy.clements@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jeclrsg** [comment]
@GordonSmith I need to go back and check for other list pages that might have this toggle in the toolbar, but wanted to check with you about the method used for the WU list page.

File: esp/src/src-react/components/Workunits.tsx [hunk 1]
```diff
@@ -217,10 +222,10 @@ export const Workunits: React.FunctionComponent<WorkunitsProps> = ({
             onClick: () => { setShowFilter(true); }
         },
         {
-            key: "mine", text: nlsHPCC.Mine, disabled: true, iconProps: { iconName: "Contact" }, canCheck: true, checked: mine,
+            key: "mine", text: nlsHPCC.Mine, disabled: !!store || !!currentUser, iconProps: { iconName: "Contact" }, canCheck: true, checked: mine,
```
Comment (GordonSmith) [inline esp/src/src-react/components/Workunits.tsx:225]: This doesn't look right to me - Should probably be `!currentUser`?

File: esp/src/src-react/hooks/user.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import * as React from "react";
+import { useConst } from "@fluentui/react-hooks";
+import { AccountService, WsAccount } from "@hpcc-js/comms";
+
+export function useCurrentUser(): { currentUser: WsAccount.MyAccountResponse } {
```
Comment (GordonSmith) [inline esp/src/src-react/hooks/user.ts:5]: Should really be called `useMyAccount`?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Review by GordonSmith — APPROVED** [review]
@jeclrsg this looks good, if you want it pulled you will need to click on "ready for review".

**Pull request merged**